### ISSUE TITLE
PointCloud generation using Continuous Autoregressive Model

### ISSUE THREAD

**Issue by KhoiDOO** [LTH14/mar#106]
Thank you for an interesting work. 

I'm trying to use this method to train a model that can generate points in 3D. I received quite good results with the random order mask sampling strategy (1024 generated points). I want to do something more that can learn the characteristics of points, which are sorted in lexical order (from low to high). 

I tried to modify the sample_orders method in the main model to get the raster order as follows:
```Python
def sample_orders(self, bsz):
  order = np.arange(self.seq_len)
  orders = np.tile(order, (bsz, 1))
  orders = torch.tensor(orders, dtype=torch.long).cuda()
  return orders
```

The result of the above function is
```Python
>>> a = sample_orders(1)
>>> a
tensor([[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]], device='cuda:0')
>>> 
```

Since the order vector is now in raster order, the model will learn to generate points causally. Was that everything that I need to do to enforce raster order learning in the model? Based on your paper, what I understand is that I do not need to use a causal mask to do so, is that right?

Thank you! I'm looking forward to your answer!
